src/hooks/useTimer.js

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useCallback, useState } from "react";
+import { MAX_SECONDS } from "../utils/constants";
+
+const interval =
+  (delay = 0) =>
+  (callback) =>
+    useEffect(() => {
+      const id = setInterval(callback, delay);
+
+      return () => clearInterval(id);
+    }, [callback]);
+const useSecondsInterval = interval(1000);
```

Comment:
refactor this logic and remove "const interval": 
const useSecondsInterval = (callback, delay = 1000) => {
  useEffect(() => {
    const id = setInterval(callback, delay);
    return () => clearInterval(id);
  }, [callback, delay]);
};